### ISSUE TITLE
[Godot 4.0] Editor Shows Error when using class_name on Script that Extends from Resource 

### ISSUE THREAD

**Issue by arisvaldez** [godotengine/godot#67727]
### Godot version

4.0.dev Beta 3

### System information

Windows 10, foward_plus, Nvidia RTX 2060

### Issue description

Editor displays message "Class Pepe(this case) hides a global script class", whenever saving a file with class_name.

In addition to this, when I create a resource based on this class and click to see it in the inspector, I cannot see the exported variables, even if I make a change it is not reflected in the inspector.

I am using Godot Editor

![class_name_error](https://user-images.githubusercontent.com/20196501/197276330-3da38509-fac2-476a-a293-ec32e215b410.png)

### Steps to reproduce

Open attached project.
Open SimpleClass.gd.
Save file
There's an error.

Create a Resource using Pepe on Search box
Double click on recent created Resource.
Look at inspector, exported variable can't use from it.
If I remove the class_name then all works fine

![class_name_error_full](https://user-images.githubusercontent.com/20196501/197277079-1c297a7c-ec64-4192-a8a0-b4fc655e0637.png)

### Minimal reproduction project

[ClassNameOnResourceExtends.zip](https://github.com/godotengine/godot/files/9842030/ClassNameOnResourceExtends.zip)

**Comment by Calinou** [comment]
Related to https://github.com/godotengine/godot/issues/67056 (possible duplicate?).

**Comment by Indieology** [comment]
I am having this same issue as well in Godot 4 beta 3

**Comment by Indieology** [comment]
I recreated my simple project from a fresh start in Godot 4 beta 4. This issue seemed to disappear. Until I pushed a commit through github of the project. Then this error reappeared. Not entirely sure this is what caused it, but seemed to be the case. @arisvaldez was this the case with you as well?

**Comment by arisvaldez** [comment]
@Indieology In my case this error appears every time I use the keyword **class_name** in a file that inherits from Resource.

every time I give save then it produces the error.

**Comment by arisvaldez** [comment]
> I recreated my simple project from a fresh start in **Godot 4 beta 4**. This issue seemed to disappear. Until I pushed a commit through github of the project. Then this error reappeared. Not entirely sure this is what caused it, but seemed to be the case. @arisvaldez was this the case with you as well?

you Mean Godot 4 beta 3?

@Indieology In my case this error appears every time I use the keyword class_name in a file that inherits from Resource.

every time I give save then it produces the error.

**Comment by arisvaldez** [comment]
> Related to #67056 (possible duplicate?).

Maybe it's the same, but in this case I added the behavior of the inspector, which happens when I try to open the resource file in Inspector

**Comment by arisvaldez** [comment]
Hi everyone!!

@Indieology , @Calinou 

I have tested the project in Godot 4 beta 6 and the error has disappeared

**Comment by andersonmadeira** [comment]
I can still see the error in Beta 14, any one else also having the issue?

**Comment by arisvaldez** [comment]
> I can still see the error in Beta 14, any one else also having the issue? EDIT: I have a Node2D (not a Resource) that has a class_name of FooBar

greetings @andersonmadeira ,  I tested on RC 2 and does not launch the error

**Comment by rabrant** [comment]
I'm having this issue extending a Resource using class_name and get the Class Hides global script class error. In Godot 4.0.1.

**Comment by fw-digital** [comment]
I also have a lot of trouble with this issue in Godot 4.0.2. It definitely is not dependent of the extended class, I get this with MeshInstance3D and others. Sometimes it helps to delete the .Godot folder to get the project running, but closing and reopening the editor raises the error again

**Comment by marcol1** [comment]
Yep, having the same issue. Can we get this fixed please? It's really annoying and actually quite a critical bug to fix as it makes inheritance far less useful.

**Comment by marcol1** [comment]
Is there a way to bump issues to gain more traction? This is pretty critical, in my opinion. Game development relies heavily on inheritance.

**Comment by andhedrew** [comment]
Same issue, only when I work on the project on a Mac.

**Comment by jean-kevin-afk** [comment]
Old issue which I can not reproduce. It has to be closed (doesn't happen in 4.1+) I think.

**Comment by Calinou** [comment]
Closing per the above comment. Please comment if you can still reproduce this on the latest Godot version.
